pathman/path.py

```
@@ -212,6 +216,237 @@ class RemotePath(object):
+        return self
+
+    def abspath(self) -> 'BlackfynnPath':
+        return self
+
+    def walk(self) -> List['BlackfynnPath']:
+        if not self.is_dir():
+            return None
+        files = []
+        queue = [(self._object, self._pathstr)]
+        while len(queue) > 0:
+            root, path = queue.pop()
+            for item in root.items:
+                item_path = os.path.join(path, item.name)
+                if 'Collection' in item.type:
+                    queue.append((item, item_path))
+                else:
+                    extension = None
+                    if hasattr(item, 'sources'):
+                        if len(item.sources) > 1:
+                            raise RuntimeError(
+                                "{} has too many sources").format(item)
+                        extension = Path(item.sources[0].s3_key).extension
+                    files.append(item_path + (extension if extension else ''))
+        return [BlackfynnPath(file) for file in files]
+
+    def glob(self, pattern: str) -> List['BlackfynnPath']:
+        regex_text = '('
+        for char in pattern:
+            if char == '*':
+                regex_text += '[^/]*'
+            elif char == '?':
+                regex_text += '.'
+            else:
+                regex_text += re.escape(char)
+        regex_text += ')'
+        regex = re.compile(regex_text)
+        files = self.walk()
+        seen = set()
+        return [file for file in files
+                for match in [regex.match(file.path)] if match
+                for path in [match.group(0)]
+                if not (path in seen or seen.add(path))]
+
+    def ls(self) -> List['BlackfynnPath']:
+        if not self.is_dir():
+            return None
+        files = []
+        for item in self._object.items:
+            ext = None
+            if hasattr(item, 'sources'):
+                ext = Path(item.sources[0].s3_key).extension
+                if len(item.sources) > 1:
```

same thing as for `walk`. I don't think I would raise a runtime error